index.js

```diff
@@ -46,7 +56,17 @@ class HdKeyring extends EventEmitter {
 
   addAccounts (numberOfAccounts = 1) {
     if (!this.root) {
-      this._initFromMnemonic(bip39.generateMnemonic())
+      if (!this.encryptionKey) {
+        throw new Error('Cannot initialize wallet without an encryption key')
+      }
+      // TODO: These chrome APIs are not yet implemented
+      // chrome.getWalletSeed permission must ONLY be granted to this extension
+      if (typeof chrome === 'object' && chrome.getWalletSeed) {
+        const seed = chrome.getWalletSeed(this.encryptionKey)
```

Comment:
I think no params needed here?

Reply:
@bbondy the encryptionKey param is needed because it is what is used to encrypt the key on disk. this is passed in by metamask since it's picked by the user. see Step 2 in https://docs.google.com/document/d/1yYzn8ycq4GzKNrd9h-DHM9Yo7R8BUez-KUGkcd2kEFE/edit after `chrome.getWalletSeed does the following in C++`